### PR TITLE
⬆️ Upgrade Camel to 3.14.0

### PR DESCRIPTION
**Pull request by zbendhiba**

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: docs/pom.xml [hunk 1]
```diff
@@ -31,6 +31,12 @@
 
     <name>Camel Quarkus :: Documentation</name>
 
+    <properties>
+        <!-- Disable xref checks as long as there is no 3.14.x branch of Camel docs -->
```
Comment (ppalaga) [inline docs/pom.xml:35]: An issue for re-enabling and pasting it's URL here would be nice.

File: docs/antora-playbook-dev.yml [hunk 2]
```diff
@@ -26,7 +26,7 @@ content:
     start_path: docs
   - url: git@github.com:apache/camel.git
     branches:
-      - camel-3.13.x # replace ${camel.docs.branch}
+      - camel-3.14.x # replace ${camel.docs.branch}
```
Comment (aldettinger) [inline docs/antora-playbook-dev.yml:29]: I thought `antora-playbook-dev.yml` and `antora-playbook` were not needed anymore with this [recent change](https://github.com/apache/camel-quarkus/commit/43cede1eca3fc257387ba448b371228df9669044).

Comment (zbendhiba) [reply docs/antora-playbook-dev.yml]: There was a problem with rebase. Thanks for catching this

File: docs/pom.xml (hunk 1 above)
Comment (zbendhiba) [reply docs/pom.xml:35]: @ppalaga  I didn't understand your comment

**Comment by zbendhiba** [comment]
I guess the staging repo is not available anymore, and artifacts not totally pushed to maven central.

**Comment by zbendhiba** [comment]
Latest push:
- removed the staging repository
- remove the @Disabled on the SQS test because the bug was fixed in 3.14.0, with CAMEL-17158
- Make CI fallback on https://repository.apache.org/content/repositories/releases, meanwhile waiting for artifacts to be available on maven central

**Comment by zbendhiba** [comment]
@ppalaga I failed setting the MAVEN_ARGS for tests running inside a sub-repo : functional-extension-tests ...

**Comment by zbendhiba** [comment]
> @ppalaga I failed setting the MAVEN_ARGS for tests running inside a sub-repo : functional-extension-tests ...
Latest build: initial install and native tests were ok
Lastest commit is an attempt to pass the absolute path of settings.xml for extension-jvm and other sub-directories maven builds

**Comment by ppalaga** [comment]
Great to see that Camel 3.14 is finally available on Central!

**Comment by zbendhiba** [comment]
The build is fine against artifacts sync in central. Merging this, to be able to prepare the release

**Pull request merged**